### PR TITLE
xenlib: add vch library headers

### PR DESCRIPTION
**Pull request by sa-kib**
Emplace initial structure of vchan implementation.

### CODE REVIEW

File: vch/include/vch.h [hunk 1]
```diff
@@ -0,0 +1,21 @@
+#pragma once
```
Comment (firscity) [inline vch/include/vch.h:1]: Please, add SPDX identifier and EPAM copyright to newly created file

**Review by firscity — CHANGES_REQUESTED** [review]
Also, please add more detailed commit message with motivation for this change.

**Comment by firscity** [comment]
Also, please look at Zephyr checkpatch output for you PR:
```
...
WARNING: Missing or malformed SPDX-License-Identifier tag in line 1
#52: FILE: vch/include/vch.h:1:
+#pragma once

WARNING: Prefer 'unsigned int' to bare use of 'unsigned'
#62: FILE: vch/include/vch.h:11:
+	unsigned write_ord;

ERROR: "(foo*)" should be "(foo *)"
#70: FILE: vch/include/vch.h:19:
+void vch_close(struct vch_handle*);
...

```

File: vch/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,7 @@
+add_library(VCH INTERFACE)
```
[the remark this answers is not in the document]
Comment (sa-kib) [reply vch/CMakeLists.txt:6]: added in 28be6bf

File: vch/include/vch.h (hunk 1 above)
Comment (sa-kib) [reply vch/include/vch.h:7]: added in 28be6bf

File: vch/include/vch.h [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+	struct vchan_interface *ring;
+	struct ring_shared *write;
+	void *write_cbuf;
+	unsigned write_ord;
+	struct ring_shared *read;
+	unsigned read_ord;
+	void *read_cbuf;
+};
+
+int vch_open(domid_t, const char *path, size_t min_rs, size_t min_ws, struct vch_handle*);
```
[the remark this answers is not in the document]
Comment (sa-kib) [reply vch/include/vch.h]: added in 28be6bf

**Comment by sa-kib** [comment]
> Also, please add more detailed commit message with motivation for this change.

commit msg updated

**Comment by firscity** [comment]
Please, add my `Reviewed-by: Dmytro Firsov <dmytro_firsov@epam.com>` and force-push squashed commit.

**Review by firscity — APPROVED** [review]

**Comment by sa-kib** [comment]
squashed & rebased

**Pull request merged**